comptests/prov/test_prov_k8s_cortx_cold_upgrade.py

```diff
@@ -0,0 +1,210 @@
+        assert_utils.assert_true(solution_path[0], solution_path[1])
+        image_dict = {"all_image": self.cortx_all_image, "rgw_image": self.cortx_rgw_image}
+        local_path = self.deploy_lc_obj.update_sol_with_image(self.local_sol_path, image_dict)
+        assert_utils.assert_true(local_path[0], local_path[1])
+        for node_obj in self.host_list:
+            resp = self.deploy_lc_obj.copy_sol_file(node_obj, local_sol_path=local_path[1],
+                                                    remote_code_path=self.
+                                                    prov_deploy_cfg["git_remote_path"])
+            assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("Step 4: Done.")
+
+        LOGGER.info("Step 5: Pull the images for upgrade.")
+        for node_obj in self.host_list:
+            for image in image_dict:
+                resp = self.deploy_lc_obj.pull_image(node_obj, image_dict[image])
+                assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("Step 5: Done.")
+
+        LOGGER.info("Step 6: Start upgrade.")
+        LOGGER.info("Upgrading CORTX image to version: %s.", self.cortx_all_image)
+        resp = self.deploy_lc_obj.upgrade_software(self.master_node_obj,
+                                                   self.prov_deploy_cfg["git_remote_path"], upgrade_type="cold")
+        assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("Step 6: Done.")
+
+        LOGGER.info("Step 7: Check if installed version is equals to installing version.")
+        resp = HAK8s.get_config_value(self.master_node_obj)
+        assert_utils.assert_true(resp[0], resp[1])
+        new_installed_version = resp[1]['cortx']['common']['release']['version']
+        LOGGER.info("New CORTX image version: %s", new_installed_version)
+        assert_utils.assert_equals(installing_version, new_installed_version,
+                                   "Installing version is not equal to new installed version.")
+        LOGGER.info("Step 7: Done.")
+        LOGGER.info("Test Completed.")
+
+    @pytest.mark.run(order=1)
+    @pytest.mark.lc
+    @pytest.mark.comp_prov
+    @pytest.mark.tags("TEST-37352")
+    def test_37352(self):
+        """
+        Verify if cortx-service or pod is in crashloopback state, cold upgrade should work.
+        """
+        LOGGER.info("Test Started.")
+        LOGGER.info("Step 1: Make sure pods are in crashloopbackoff state.")
+        resp = self.master_node_obj.execute_cmd(commands.CMD_POD_STATUS, read_lines=True)
+        LOGGER.info(type(resp))
+        for output in resp:
+                output = output.split("\n")
+                resp = str(resp)
+                if("Init:ImagePullBackOf") in output[0]:
+                    LOGGER.info(output)
+                    LOGGER.info("Step 1: Done.")
+        
+                    LOGGER.info("Step 2: Get installed version.")
```

Comment:
Are all the steps expected to be part of for loop ?

Reply:
Yes